app/assets/javascripts/homeroom_table/HomeroomTable.js

```diff
@@ -4,36 +4,29 @@ import SortHelpers from '../helpers/sort_helpers.jsx';
 import Cookies from 'js-cookie';
 
 
-export default React.createClass({
-  displayName: 'HomeroomTable',
+class HomeroomTable extends React.Component {
 
-  propTypes: {
-    showStar: React.PropTypes.bool.isRequired,
-    showMcas: React.PropTypes.bool.isRequired,
-    rows: React.PropTypes.array.isRequired
-  },
-
-  getInitialState () {
+  constructor(props) {
+    super(props);
     const initialColumns = this.getInitialColumnsDisplayed();
-
-    return {
+    this.state = {
       columnsDisplayed: initialColumns,
       showColumnPicker: false,
       sortBy: 'first_name',
       sortType: 'string',
       sortDesc: true
     };
-  },
+  }
 
-  orderedStudents () {
+  orderedStudents() {
```

Comment:
@alexsoble This js file has minimum change. The only changes are the `,` and moving in `()` to the function name. Nothing has been removed. All of the functions in here are closed correctly. I double check the `constructor` with `state` getting set.